ci/ci_helpers.sh

```diff
@@ -22,11 +22,11 @@ hint_on_fail() {
     exit_code=$?
     # Get filename of the currently running script
     # Source: https://stackoverflow.com/a/192337/2570866
-    filename=$(basename "$0")
+    filename=$(basename "$0" | sed 's/\.sh$/sh/g')
```

Comment:
Why are we replacing .sh -> sh?

I think using a different delimiter could increase readability here
`s#\.sh$#sh#g`

Reply:
because we put it in the anchor and the anchor has the `.` stripped on github. I'll rename the variable to be anchor.

Reply:
Added some comments and reorderd the code a bit to make it clearer. I also used bash replacement syntax now, because `shellcheck` suggested it now.